### PR TITLE
Reworking `mkDebounce`

### PR DESCRIPTION
**Pull request by Vlix**
Ok, so I've tried to rework the debouncing, because of issue #980, but it seems to fail on a test because the trigger `MVar` doesn't seem to be correctly filled unless I put a `trace` in a specific spot!?

Does anyone know how to force/evaluate (or whatever is happening here) the `tryPutMVar` without using `trace`?

- auto-update: reworked 'mkDebounce', and it should work... but 'Leading' only works if I put a 'trace' in a specific spot???

---

Still TODO:

- [x] Bumped the version number

After submitting your PR:

- [x] Update the Changelog.md file with a link to your PR
- [x] Check that CI passes (or if it fails, for reasons unrelated to your change, like CI timeouts)

### CODE REVIEW

**Comment by Vlix** [comment]
Hmmm, seeing the random failing in CI jobs, this is probably a timing issue? And the `trace "..."` printing to the screen is just takes enough time with the GHC version I was developing with? 🤔 

Now to figure out if the test should be adjusted, or if there is actually an error in the implementation...

**Comment by Vlix** [comment]
I figured it out. It was the `trigger` MVar that should be emptied BEFORE forking.

I think this fixes the leaking of threads and adds two styles of debouncing that are also pretty common.

**Comment by Vlix** [comment]
Windows nightly fails with:
`doctest > C:/Users/runneradmin/AppData/Local/Programs/stack/x86_64-windows/ghc-9.8.3/mingw/bin/strip.exe: error: Permission denied`

... who's fault would this be? :thinking:

**Comment by Vlix** [comment]
Oh god, `data-default` breaks literally every CI... why is this even still a thing!?

**Comment by kazu-yamamoto** [comment]
@Vlix Is this PR ready for reviewing?

**Comment by Vlix** [comment]
@kazu-yamamoto Yes, I hope with this last commit (and `quic` building again) that CI is back to working condition.

**Comment by Vlix** [comment]
Apparently my new `auto-update` is a bit flaky :/

**Comment by Vlix** [comment]
When this is merged, I'll update the CI in #1011 to include `quic` again.

File: auto-update/Control/Debounce.hs [hunk 1]
```diff
@@ -23,23 +23,27 @@
...
+
+    -- * Settings
     DI.DebounceSettings,
     defaultDebounceSettings,
 
-    -- * Accessors
+    -- ** Accessors
     DI.debounceFreq,
     DI.debounceAction,
     DI.debounceEdge,
```
Comment (kazu-yamamoto) [inline auto-update/Control/Debounce.hs:36]: I wonder why `DebounceEdge` is NOT exported while `deboundeEdge` is exported.

Comment (kazu-yamamoto) [reply auto-update/Control/Debounce.hs:36]: Never mind.
`leadingEdge` should be used instead of `Leading`.

**Comment by kazu-yamamoto** [comment]
Rebased and merged.
Nice work!

**Pull request closed without merging**